src/commands/containers/browseContainer.ts

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { IActionContext, TelemetryProperties } from "vscode-azureextensionui";
+import { ext } from "../../extensionVariables";
+import { ContainerTreeItem } from "../../tree/containers/ContainerTreeItem";
+import { captureCancelStep } from '../../utils/captureCancelStep';
+
+type BrowseTelemetryProperties = TelemetryProperties & { possiblePorts?: number[], selectedPort?: number };
+
+type ConfigureBrowseCancelStep = 'node' | 'port';
+
+async function captureBrowseCancelStep<T>(cancelStep: ConfigureBrowseCancelStep, properties: BrowseTelemetryProperties, prompt: () => Promise<T>): Promise<T> {
+    return await captureCancelStep(cancelStep, properties, prompt)
+}
+
+// NOTE: These ports are ordered in order of preference.
+const commonWebPorts = [
+    443,    // SSL
+    80,     // HTTP
+    3000,   // (Node.js) Express.js
+    3001,   // (Node.js) Sails.js
+    5001,   // (.NET Core) ASP.NET SSL
+    5000,   // (.NET Core) ASP.NET HTTP
+    8080,   // (Node.js)
+    8081    // (Node.js)
+];
+
+const commonSslPorts = [
+    443,    // SSL
+    5001    // (.NET Core) ASP.NET SSL
+];
+
+function parsePortAndProtocol(portAndProtocol: string): { port: number, protocol: string | undefined } | undefined {
+    const splitPortAndProtocol = portAndProtocol.split('/');
+
+    if (splitPortAndProtocol.length > 0) {
+        const port = Number.parseInt(splitPortAndProtocol[0], 10);
+
+        if (!Number.isNaN(port)) {
+            return {
+                port,
+                protocol: splitPortAndProtocol.length > 1 ? splitPortAndProtocol[1].toLowerCase() : undefined
+            };
+        }
+    }
+
+    return undefined;
+}
+
+export async function browseContainer(context: IActionContext, node?: ContainerTreeItem): Promise<void> {
+    const telemetryProperties = <BrowseTelemetryProperties>context.telemetry.properties;
+
+    if (!node) {
+        node = await captureBrowseCancelStep('node', telemetryProperties, () => ext.containersTree.showTreeItemPicker<ContainerTreeItem>(ContainerTreeItem.allContextRegExp, context));
+    }
+
+    const inspectInfo = await node.getContainer().inspect();
+
+    const ports = inspectInfo.NetworkSettings && inspectInfo.NetworkSettings.Ports || {};
```

Comment:
Though I'm pretty sure `inspectInfo` and `NetworkSettings` will always be present, should we do `inspectInfo && inspectInfo.NetworkSettings && ...` just to be safe?

Reply:
Couldn't hurt.